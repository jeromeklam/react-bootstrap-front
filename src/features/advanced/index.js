export * from './DefaultForm.jsx';
export { default as DefaultQuickSearch } from './DefaultQuickSearch.jsx';
export { default as HoverObserver } from './HoverObserver.jsx';
export * from './MobileForm.jsx';
export { default as MobileQuickSearch } from './MobileQuickSearch.jsx';
export { default as Portal } from './Portal.jsx';
export * from './ResponsiveConfirm.jsx';
export * from './ResponsiveForm.jsx';
export * from './ResponsiveModal.jsx';
export { default as ResponsiveModalInner } from './ResponsiveModalInner.jsx';
export * from './ResponsiveQuickSearch.jsx';
export { default as WidthObserver } from './WidthObserver.jsx';
export { default as SvgMask } from './SvgMask';
export { default as Follower } from './Follower';
export { default as SvgTimer } from './SvgTimer';
export { default as SvgPlay } from './SvgPlay';
