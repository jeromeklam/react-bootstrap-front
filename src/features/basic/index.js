export { default as Dropdown } from "./Dropdown";
export * from "./InlineInputCheckbox";
export * from "./InlineInputPicker";
export { default as InlineInputSelect } from "./InlineInputSelect";
export * from "./InlineInputText";
export * from "./InlineLabel";
export * from "./InputCheckbox";
export { default as InputDateWeb } from "./InputDateWeb";
export { default as InputDatetime } from "./InputDatetime";
export * from "./InputEmail";
export * from "./InputHidden";
export { default as InputMask } from "./InputMask";
export { default as InputMonetary } from "./InputMonetary";
export * from "./InputPassword";
export { default as InputPicker } from "./InputPicker";
export { default as ButtonPicker } from "./ButtonPicker";
export * from "./InputRadio";
export * from "./InputRandomText";
export { default as InputSelect } from "./InputSelect";
export { default as InputSpin } from "./InputSpin";
export { default as InputStringarray } from "./InputStringarray";
export * from "./InputText";
export { default as InputTextarea } from "./InputTextarea";
export * from "./ResponsiveButton";
export { default as InputCheckList } from "./InputCheckList";
export { default as CalmButton } from "./CalmButton";
export { default as InputGroup } from "./InputGroup";
export { default as InputGroupAppend } from "./InputGroupAppend";
export { default as InputGroupPrepend } from "./InputGroupPrepend";
export { default as InputGroupText } from "./InputGroupText";
export { default as DropdownMenu } from "./DropdownMenu";
export { default as DropdownMenuHeader } from "./DropdownMenuHeader";
export { default as DropdownMenuOption } from "./DropdownMenuOption";
export { default as DropdownMenuDivider } from "./DropdownMenuDivider";
export { default as InputKeyword } from "./InputKeyword";
export { default as InputKeywords } from "./InputKeywords";
export { default as InputWorkload } from "./InputWorkload";
export { default as InputGpsCoords } from "./InputGpsCoords";
export { default as InputPhone } from "./InputPhone";
export { default as Badge } from "./Badge";
export { default as InputDate } from "./InputDate";
export * from "./InputDateNatif";
export { default as InputPhoto } from "./InputPhoto";
export { default as InputBarCode } from "./InputBarCode";
export { default as Scanner } from "./Scanner";
export { default as InputSignature } from "./InputSignature";
export * from "./InputTextMultiline";
export * from "./InputChildren";
