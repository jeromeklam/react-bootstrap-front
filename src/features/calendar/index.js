import EventWrapper from './EventWrapper'
import BackgroundWrapper from './BackgroundWrapper'

export const components = {
  eventWrapper: EventWrapper,
  timeSlotWrapper: BackgroundWrapper,
  dateCellWrapper: BackgroundWrapper,
}

export { default as Calendar } from './Calendar'
export { DateLocalizer } from './localizer'
export { default as momentLocalizer } from './moment'
export { default as globalizeLocalizer } from './globalize'
export { default as dateFnsLocalizer } from './date-fns'
export { default as move } from './utils/move'
export { views as Views, navigate as Navigate } from './utils/constants'
export { default as Agenda } from './Agenda';
export { default as BackgroundCells } from './BackgroundCells';
export { default as DateContentRow } from './DateContentRow';
export { default as DateHeader } from './DateHeader';
export { default as Day } from './Day';
export { default as DayColumn } from './DayColumn';
export { default as EventCell } from './EventCell';
export { default as EventEndingRow } from './EventEndingRow';
export { default as EventRow } from './EventRow';
export { default as EventRowMixin } from './EventRowMixin';
export { default as EventWrapper } from './EventWrapper';
export { default as Header } from './Header';
export { default as IconWrapper } from './IconWrapper';
export { default as Month } from './Month';
export { default as NoopWrapper } from './NoopWrapper';
export { default as Popup } from './Popup';
export { default as ResourceHeader } from './ResourceHeader';
export { default as Selection } from './Selection';
export { default as TimeGrid } from './TimeGrid';
export { default as TimeGridEvent } from './TimeGridEvent';
export { default as TimeGridHeader } from './TimeGridHeader';
export { default as TimeGutter } from './TimeGutter';
export { default as TimeSlotGroup } from './TimeSlotGroup';
export { default as Toolbar } from './Toolbar';
export { default as Week } from './Week';
export { default as WorkWeek } from './WorkWeek';
