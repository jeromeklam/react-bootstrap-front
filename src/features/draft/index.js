export { default as ColorPicker } from "./ColorPicker";
export { default as FontSize } from "./FontSize";
export { default as Inline } from "./Inline";
export { default as InputHtml } from "./InputHtml";