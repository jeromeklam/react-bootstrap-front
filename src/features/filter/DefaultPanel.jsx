import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { Filter, FilterBuilder, FILTER_SEARCH_SIMPLE } from '../filter';
import { SortableList } from '../sort';

const sortLocal = (cols, sort) => {
  let local = [];
  sort.forEach(elt => {
    let elt3 = false;
    cols.forEach(elt2 => {
      if (elt.col === elt2.col) {
        elt3 = elt2;
        return true;
      }
    });
    let newElt = {
      col: elt.col,
      way: elt.way,
      label: elt3.label,
      full: elt3,
    };
    local.push(newElt);
  });
  cols.forEach(elt => {
    if (elt.sortable) {
      const found = local.find(elt2 => {
        return elt2.col === elt.col;
      });
      if (!found) {
        let newElt = {
          col: elt.col,
          way: 'none',
          label: elt.label,
          full: elt,
        };
        local.push(newElt);
      }
    }
  });
  return local;
};

const mystyle = {
  position: 'absolute',
  left: '0px',
  right: '0px',
  top: '0px',
  bottom: '0px',
};

const innerstyle = {
  position: 'absolute',
  left: '0px',
  right: '0px',
  top: '50px',
  bottom: '0px',
  overflowX: 'hidden',
  overflowY: 'auto',
};

const btStyle = {
  position: 'absolute',
  right: '2px',
  top: '2px',
};

export default class DefaultPanel extends Component {
  static propTypes = {
    cols: PropTypes.element.isRequired,
    filters: PropTypes.element.isRequired,
    validPanelIcon: PropTypes.element.isRequired,
    cancelPanelIcon: PropTypes.element.isRequired,
    onToggleFilter: PropTypes.func.isRequired,
    sortNoneIcon: PropTypes.element,
    sortUpIcon: PropTypes.element,
    sortDownIcon: PropTypes.element,
    simpleMode: PropTypes.bool,
  };

  static defaultProps = {
    sortNoneIcon: '',
    sortUpIcon: '',
    sortDownIcon: '',
    simpleMode: false,
  };

  static getDerivedStateFromProps(props, state) {
    if (props.filters && !state.local) {
      if (props.filters !== state.filters) {
        let filters = new Filter();
        if (props.filters) {
          filters = props.filters.clone();
        }
        return { filter: filters, local: false };
      }
    }
    return { local: false };
  }

  constructor(props) {
    super(props);
    let filters = new Filter();
    if (props.filters) {
      filters = props.filters.clone();
    }
    this.state = {
      current: props.sort,
      panel: 'filter',
      sort: sortLocal(props.cols, props.sort),
      filter: filters,
      currentFilter: filters,
      local: false,
    };
    this.changePanel = this.changePanel.bind(this);
    this.onFilterChange = this.onFilterChange.bind(this);
    this.onOperator = this.onOperator.bind(this);
    this.onFilterMode = this.onFilterMode.bind(this);
    this.onFilterOperator = this.onFilterOperator.bind(this);
    this.onSortEnd = this.onSortEnd.bind(this);
    this.onSortChange = this.onSortChange.bind(this);
    this.onValid = this.onValid.bind(this);
  }

  onFilterChange(event, oper = false) {
    let { filter } = this.state;
    filter.addFilter(event.target.name, event.target.value, oper);
    filter.setSearch(FILTER_SEARCH_SIMPLE);
    this.setState({ filter, local: true });
  }

  onFilterMode(event) {
    let { filter } = this.state;
    filter.setMode(event.target.value);
    filter.setSearch(FILTER_SEARCH_SIMPLE);
    this.setState({ filter, local: true });
  }

  onOperator(event) {
    let { filter } = this.state;
    filter.setOperator(event.target.value);
    filter.setSearch(FILTER_SEARCH_SIMPLE);
    this.setState({ filter, local: true });
  }

  onFilterOperator(event) {
    let { filter } = this.state;
    const col = event.target.name;
    const oper = event.target.value;
    filter.updateFilterOperator(col.replace('oper-', ''), oper);
    this.setState({ filter, local: true });
  }

  onSortEnd({ oldIndex, newIndex }) {
    let old = this.state.sort[oldIndex];
    let way = old.way;
    if (way === 'none') {
      way = 'up';
    }
    old.way = way;
    let newSort = this.state.sort;
    if (newIndex > oldIndex) {
      newSort.splice(newIndex + 1, 0, old);
      newSort.splice(oldIndex, 1);
    } else {
      newSort.splice(newIndex, 0, old);
      newSort.splice(oldIndex + 1, 1);
    }
    this.setState({ sort: newSort });
  }

  onSortChange(col) {
    let newSort = this.state.sort;
    newSort.forEach(elt => {
      if (elt.col === col.col) {
        if (elt.way === 'none') {
          elt.way = 'up';
        } else {
          if (elt.way === 'up') {
            elt.way = 'down';
          } else {
            elt.way = 'none';
          }
        }
        return true;
      }
    });
    this.setState({ sort: newSort });
  }

  onValid(event) {
    if (event) {
      event.preventDefault();
    }
    let sort = [];
    this.state.sort.forEach(elt => {
      if (elt.way !== 'none') {
        const nElt = {
          col: elt.col,
          way: elt.way,
        };
        sort.push(nElt);
      }
    });
    this.props.onToggleFilter(this.state.filter, sort);
  }

  changePanel(panel) {
    this.setState({ panel });
  }

  render() {
    return (
      <div className="default-list-panel-inner" style={mystyle}>
        <div className={classnames('default-list-panel-navbar clearfix')}>
          <div className="common-responsive-list-panels-close btn-group" style={btStyle}>
            {!this.props.simpleMode && (
              <button
                className="btn btn-light btn-outline-secondary-light text-secondary"
                onClick={this.props.onToggleFilter}
              >
                {this.props.cancelPanelIcon}
              </button>
            )}
            <button className="btn btn-primary text-light" onClick={this.onValid}>
              {this.props.validPanelIcon}
            </button>
          </div>
          {!this.props.simpleMode && (
            <ul className="nav nav-tabs float-left">
              <li className="nav-item">
                <a
                  className={classnames(
                    'nav-link',
                    this.state.panel === 'filter' ? 'bg-secondary text-light' : 'text-secondary'
                  )}
                  onClick={() => {
                    this.changePanel('filter');
                  }}
                >
                  Filtres
                </a>
              </li>
              <li className="nav-item">
                <a
                  className={classnames(
                    'nav-link',
                    this.state.panel === 'sort' ? 'bg-secondary text-light' : 'text-secondary'
                  )}
                  onClick={() => {
                    this.changePanel('sort');
                  }}
                >
                  Tri
                </a>
              </li>
            </ul>
          )}
        </div>
        <div className="custom-scrollbar" style={innerstyle}>
          <div className="default-list-panel-content">
            {(this.props.simpleMode || this.state.panel === 'filter') && (
              <div className="default-list-panel-tab p-2">
                <FilterBuilder
                  {...this.props}
                  filters={this.state.filter}
                  onChange={this.onFilterChange}
                  onMode={this.onFilterMode}
                  onOperator={this.onOperator}
                  onFilterOperator={this.onFilterOperator}
                />
              </div>
            )}
            {!this.props.simpleMode && this.state.panel === 'sort' && (
              <div className="default-list-panel-tab p-2">
                <SortableList
                  items={this.state.sort}
                  onSortEnd={this.onSortEnd}
                  onSortChange={this.onSortChange}
                  sortNoneIcon={this.props.sortNoneIcon}
                  sortUpIcon={this.props.sortUpIcon}
                  sortDownIcon={this.props.sortDownIcon}
                  pressDelay={200}
                />
              </div>
            )}
          </div>
        </div>
      </div>
    );
  }
}
