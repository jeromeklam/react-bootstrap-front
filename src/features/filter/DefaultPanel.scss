.default-list-panel-inner {
  border: 1px solid;
  border-color: theme-color('secondary-light');
}
.default-list-panel-header {
  position: absolute;
  top: 0px;
  left: 0px;
  right: 0px;
  height: 50px;

  span {
    position: absolute;
    font-size: 1.2rem;
    font-weight: bold;
    margin-left: 15px;
    margin-top: 10px;
  }
}
.default-list-panel-navbar {
  position: absolute;
  top: 50px;
  left: 0px;
  right: 0px;
  padding-top: 4px;
  border-bottom: 1px solid;
}
