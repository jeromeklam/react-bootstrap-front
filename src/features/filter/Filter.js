import {
  FILTER_MODE_OR,
  FILTER_OPER_LIKE,
  FILTER_OPER_EMPTY,
  FILTER_OPER_NOT_EMPTY,
  FILTER_TYPE_GROUP,
  FILTER_TYPE_ELEM,
  FILTER_SEARCH_NONE,
} from './';

/**
 * Manage filters
 */
export default class Filter {
  constructor() {
    this.data = {
      operator: FILTER_OPER_LIKE,
      mode: FILTER_MODE_OR,
      origs: [],
      filters: [],
      filter: FILTER_TYPE_GROUP,
      filter_name: '',
      filter_fixed: false,
      filter_default: false,
      filter_enable: true,
      filter_crits: [],
      search: FILTER_SEARCH_NONE,
    };
    this.init();
  }

  init(mode = FILTER_MODE_OR, operator = FILTER_OPER_LIKE) {
    const { origs } = this.data || [];
    let newFilters = [];
    origs.forEach(elem => {
      if (elem.isFixed() || (elem.isDefault() && elem.isEnable())) {
        newFilters.push(elem);
      }
    });
    this.data.mode = mode;
    this.data.operator = operator;
    this.data.filters = newFilters;
  }

  setOperator(oper, all = true) {
    this.data.operator = oper;
    if (all) {
      this.data.filters.forEach(elt => {
        elt.setOperator(oper);
      });
    }
  }

  getOperator() {
    if (this.data.operator !== false) {
      return this.data.operator;
    }
    return FILTER_OPER_LIKE;
  }

  isFixed() {
    return this.data.filter_fixed;
  }

  isDefault() {
    return this.data.filter_default;
  }

  isEnable() {
    return this.data.filter_enable;
  }

  setMode(mode) {
    this.data.mode = mode;
  }

  getMode() {
    return this.data.mode;
  }

  setDefault(def = true) {
    this.data.filter_default = def;
  }

  setEnable(ena = true) {
    this.data.filter_enable = ena;
  }

  setFilterName(name, fixed = false, def = false, enable = true) {
    this.data.filter_name = name;
    this.data.filter_fixed = fixed;
    this.data.filter_default = def;
    this.data.filter_enable = enable;
    this.data.filter = FILTER_TYPE_ELEM;
  }

  getFilterName() {
    return this.data.filter_name;
  }

  setFilterCrit(value, oper = false) {
    if (oper !== false) {
      this.data.operator = oper;
    }
    this.data.filter_crits.default = value;
  }

  getFilterCrit() {
    return this.data.filter_crits.default || null;
  }

  getFilterCrits() {
    return this.data.filter_crits;
  }

  setSearch(search) {
    this.data.search = search;
  }

  getSearch() {
    return this.data.search;
  }

  checkFilters() {
    let filters = [];
    this.data.filters.forEach(elt => {
      let add = false;
      if (elt && elt.data && (elt.data.operator === FILTER_OPER_EMPTY || elt.data.operator === FILTER_OPER_NOT_EMPTY)) {
        add = true;
      } else {
        const crits3 = elt.getFilterCrits();
        Object.keys(crits3).forEach(key => {
          if (crits3[key] !== '') {
            add = true;
          }
        });
      }
      if (add) {
        filters.push(elt);
      }
    });
    this.data.filters = filters;
  }

  addFilter(name, value, oper = false, fixed = false, def = false, enable = true) {
    let elem = this.data.filters.find(elt => elt.getFilterName() === name);
    if (elem) {
      elem.setFilterCrit(value, oper);
    } else {
      let elt2 = new Filter();
      elt2.setFilterName(name, fixed, def, enable);
      elt2.setFilterCrit(value, oper);
      this.data.filters.push(elt2);
    }
    if (fixed || def) {
      let elem = this.data.origs.find(elt => elt.getFilterName() === name);
      if (elem) {
        elem.setFilterCrit(value, oper);
      } else {
        let elt2 = new Filter();
        elt2.setFilterName(name, fixed, def, enable);
        elt2.setFilterCrit(value, oper);
        this.data.origs.push(elt2);
      }
    }
    this.checkFilters();
  }

  updateFilterOperator(name, oper = false) {
    let elem = this.data.filters.find(elt => elt.getFilterName() === name);
    if (elem) {
      elem.setOperator(oper);
    } else {
      let elt2 = new Filter();
      elt2.setFilterName(name);
      if (oper !== false) {
        elt2.setOperator(oper);
      }
      this.data.filters.push(elt2);
    }
  }

  findFirst(name) {
    return this.data.filters.find(elt => elt.getFilterName() === name && elt.isEnable());
  }

  asJsonApiObject() {
    let params = {};
    let crits = {};
    let fixed = {};
    if (this.data.filters.length > 0) {
      params.filter = {};
      this.data.filters.forEach(elt => {
        let crits2 = [];
        const crits3 = elt.getFilterCrits();
        Object.keys(crits3).forEach(key => {
          if (crits3[key] === null || crits3[key] === '') {
            if (elt.getOperator() === 'empty' || elt.getOperator() === 'nempty') {
              crits2[elt.getOperator()] = crits3[key];
            }
          } else {
            crits2[elt.getOperator()] = crits3[key];
          }
        });
        if (crits2 === []) {
          if (elt.getOperator() === 'empty' || elt.getOperator() === 'nempty') {
            crits2[elt.getOperator()] = null;
          }
        }
        if (elt.isEnable()) {
          if (elt.isFixed() || elt.isDefault()) {
            fixed[elt.getFilterName()] = crits2;
          } else {
            crits[elt.getFilterName()] = crits2;
          }
        }
      });
      if (fixed !== {}) {
        if (this.data.mode === FILTER_MODE_OR) {
          params.filter = {
            and: {
              or: crits,
            },
            ...fixed,
          };
        } else {
          params.filter.and = { ...crits, ...fixed };
        }
      } else {
        if (this.data.mode === FILTER_MODE_OR) {
          params.filter.or = crits;
        } else {
          params.filter.and = crits;
        }
      }
    }
    return params;
  }

  clone() {
    let newFilter = new Filter();
    const datas = this.data;
    newFilter.data = { ...datas };
    return newFilter;
  }

  isEmpty() {
    let empty = true;
    if (this.data.filters.length > 0) {
      const found = this.data.filters.find(elem => elem.isFixed() === false && elem.isEnable() === true);
      if (found) {
        empty = false;
      }
    }
    return empty;
  }

  isDefaultOnly() {
    let defaultEnableOnly = false;
    if (this.data.filters.length > 0) {
      if (this.data.filters.find(elem => elem.isDefault() === true)) {
        defaultEnableOnly = true;
        const found = this.data.filters.find(
          elem =>
            elem.isFixed() === false &&
            (elem.isDefault() === false || (elem.isDefault() === true && elem.isEnable() === false))
        );
        if (found) {
          defaultEnableOnly = false;
        }
      }
    }
    return defaultEnableOnly;
  }

  isDefaultExist() {
    let filterDefault = false;
    if (this.data.origs.length > 0) {
      const found = this.data.origs.find(elem => elem.isDefault() === true);
      if (found) {
        filterDefault = true;
      }
    }
    return filterDefault;
  }

  disableDefaults() {
    const { origs } = this.data || [];
    origs.forEach(elem => {
      if (elem.isDefault()) {
        elem.setEnable(false);
      }
    });
  }

  enableDefaults() {
    const { origs } = this.data || [];
    origs.forEach(elem => {
      if (elem.isDefault()) {
        elem.setEnable(true);
      }
    });
  }
}
