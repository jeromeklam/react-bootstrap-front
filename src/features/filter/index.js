/* MODES */
export const FILTER_MODE_OR = 'OR';
export const FILTER_MODE_AND = 'AND';
/* OPERATORS */
export const FILTER_OPER_LIKE = 'contains';
export const FILTER_OPER_SOUND_LIKE = 'soundex';
export const FILTER_OPER_NOT_LIKE = 'ncontains';
export const FILTER_OPER_START_WITH = 'containsb';
export const FILTER_OPER_END_WITH = 'containse';
export const FILTER_OPER_EQUAL = 'eq';
export const FILTER_OPER_NOT_EQUAL = 'neq';
export const FILTER_OPER_EMPTY = 'empty';
export const FILTER_OPER_NOT_EMPTY = 'nempty';
export const FILTER_OPER_LOWER = 'ltw';
export const FILTER_OPER_LOWER_OR_EQUAL = 'ltwe';
export const FILTER_OPER_LOWER_OR_EQUAL_OR_NULL = 'ltwen';
export const FILTER_OPER_GREATER = 'gt';
export const FILTER_OPER_GREATER_OR_EQUAL = 'gte';
export const FILTER_OPER_GREATER_OR_EQUAL_OR_NULL = 'gten';
export const FILTER_OPER_IN = 'in';
export const FILTER_OPER_NOT_IN = 'nin';
export const FILTER_OPER_BETWEEN = 'between';
/* TYPES */
export const FILTER_TYPE_GROUP = 'GROUP';
export const FILTER_TYPE_ELEM = 'ELEM';
/* SEARCH MODE */
export const FILTER_SEARCH_NONE = 'NONE';
export const FILTER_SEARCH_QUICK = 'QUICK';
export const FILTER_SEARCH_SIMPLE = 'SIMPLE';

export { default as DefaultPanel } from './DefaultPanel.jsx';
export { default as Filter } from './Filter.js';
export { default as FilterBuilder } from './FilterBuilder.jsx';
export { default as InputAutocomplete } from './InputAutocomplete';
export { default as FilterHeader } from './FilterHeader.jsx';
