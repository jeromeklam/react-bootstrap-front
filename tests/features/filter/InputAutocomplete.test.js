import React from 'react';
import renderer from 'react-test-renderer';
import { InputAutocomplete } from '../../../src/features/filter';

const callback = jest.fn();

it('renders standard', () => {
  // First render
  const component = renderer.create(
    <InputAutocomplete id="test" name="test" type="TEST" list={[]} clearIcon={<span />} onSearch={callback} onSelect={callback} />
  );
  let test = component.toJSON();
  expect(test).toMatchSnapshot();
});
