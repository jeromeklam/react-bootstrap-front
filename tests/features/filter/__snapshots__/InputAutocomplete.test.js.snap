// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders standard 1`] = `
<div
  className="form-group"
>
  
  <div
    className=""
  >
    <div
      className="input-group"
    >
      <input
        autoComplete="off"
        className="border-primary form-control"
        disabled={false}
        id="test"
        name="test"
        onChange={[Function]}
        onClick={[Function]}
        onKeyUp={[Function]}
        required={false}
        type="text"
        value=""
      />
      <div
        className="input-group-append"
      >
        <button
          className="btn btn-input btn-outline-primary bg-light"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <span />
        </button>
      </div>
    </div>
  </div>
</div>
`;
